src/Microsoft.AspNetCore.SignalR/HubEndPoint.cs

```diff
@@ -59,24 +59,54 @@ public class HubEndPoint<THub, TClient> : IInvocationBinder where THub : Hub<TCl
 
         public async Task OnConnectedAsync(ConnectionContext connection)
         {
-            await ProcessNegotiate(connection);
+            var output = Channel.CreateUnbounded<byte[]>();
+            var connectionContext = new HubConnectionContext(output, connection);
+
+            await ProcessNegotiate(connectionContext);
```

Comment:
It's not negotiation, that's a misnomer. They client tells the sever what protocol it supports, the sever either has that exact protocol or it throws.

Seems like we're only talking about websockets here. The sever has settting per endpoint that determines what to send (binary or text) and there's no way to do this in a per connection level. Maybe that's what you mean? Assume we have a way to set that in some connection metadata or feature so that it can be read by the websocket transport.

For SSE why would the client bother to send a request to the server if we ended up with an invalid combination, say "protobuf and SSE" or msgpack and SSE? 

Were we going to support both http transport and hub protocol negotiation?